src/mongocxx/client.hpp

```diff
@@ -61,7 +68,7 @@ class MONGOCXX_API client {
     ///   Additional options that cannot be specified via the mongodb_uri
     ///
     client(
-        const class uri& mongodb_uri = mongocxx::uri(),
+        const class uri& mongodb_uri,
```

Comment:
Yeah, this one could go either way. I think that outside of examples and tests though, almost all real uses of the driver will pass a non-default constructed uri argument, since they won't actually want to connect to localhost. So, maybe it won't be that strange in practice? Another option would be to eliminate the default constructor for URI and require providing a string? That way it wouldn't look quite so odd. Open to suggestions on this one.